semantic/constraints.go

```diff
@@ -306,14 +306,33 @@ func (v ConstraintGenerator) typeof(n Node) (PolyType, error) {
 		if !ok {
 			return nil, errors.New("member object must be a type variable")
 		}
-		v.cs.AddKindConst(tv, KRecord{
+		v.cs.AddKindConst(tv, ObjectKind{
 			properties: map[string]PolyType{n.Property: ptv},
 			lower:      LabelSet{n.Property},
 			upper:      AllLabels(),
 		})
 		return ptv, nil
+	case *IndexExpression:
+		ptv := v.cs.f.Fresh()
+		t, err := v.lookup(n.Object)
+		if err != nil {
+			return nil, err
+		}
+		tv, ok := t.(Tvar)
+		if !ok {
+			return nil, errors.New("object must be a type variable")
```

Comment:
this should read `array must be a type variable`.

Reply:
Ah yes I forgot to change that.